### ISSUE TITLE
Dota 2 Client keeps crashing [Mac OS X]

### ISSUE THREAD

**Issue by Th3NightHawk** [ValveSoftware/Dota-2#1171]
My Dota 2 client keeps crashing without any errors once I launch into a game. It's fine in the menu but even going into the Demo Hero mode crashes the game. I'm not sure where I can get crash logs for you but the below gets logged in the DiagnosticReports folder. There is a .gpuRestart file for each time a crash occurred. Below is the content:
```
Thu Feb  2 23:07:39 2017

Event:               GPU Reset
Date/Time:           Thu Feb  2 23:07:39 2017
Application:         dota2
Path:                
GPUSubmission Trace ID: 1b000a2cba
OS Version:          Mac OS X Version 10.12.4 (Build 16E144f)
Graphics Hardware:   Intel Iris Pro
Signature:           8013

Report Data:

===========================================================================
                         Intel GPU Hang Summary
                              version 1.0.0
===========================================================================

-- Summary of Status for ALL Rings --
MAIN Ring	is hung in a batch buffer	 <-- ROOT CAUSE!
MEDIA Ring	is idle
BLT Ring	is waiting on a semaphore
VEBOX Ring	is idle
===========================================================================
                     Essential Ring Registers
===========================================================================

-- MAIN Ring --
  RING_HEAD   = 0x01068f4c  (offset = 0x00068f4c, wrap count = 8, wait_for_cond = 0x0)
  RING_TAIL   = 0x0006e020  (offset = 0x0006e020)
  RING_START  = 0x10205000  (offset = 0x10205000)
  RING_CTRL   = 0x000ff001  (enabled, auto report disabled (0x00), not waiting, semaphore not waiting, length = 0x100 4KB pages)

-- MEDIA Ring --
  RING_HEAD   = 0x00000168  (offset = 0x00000168, wrap count = 0, wait_for_cond = 0x0)
  RING_TAIL   = 0x00000168  (offset = 0x00000168)
  RING_START  = 0x10325000  (offset = 0x10325000)
  RING_CTRL   = 0x0000f001  (enabled, auto report disabled (0x00), not waiting, semaphore not waiting, length = 0x010 4KB pages)

-- BLT Ring --
  RING_HEAD   = 0x0e6118d8  (offset = 0x000118d8, wrap count = 115, wait_for_cond = 0x0)
  RING_TAIL   = 0x00012610  (offset = 0x00012610)
  RING_START  = 0x10305000  (offset = 0x10305000)
  RING_CTRL   = 0x0001f401  (enabled, auto report disabled (0x00), not waiting, semaphore is waiting, length = 0x020 4KB pages)

-- VEBOX Ring --
  RING_HEAD   = 0x000000f0  (offset = 0x000000f0, wrap count = 0, wait_for_cond = 0x0)
  RING_TAIL   = 0x000000f0  (offset = 0x000000f0)
  RING_START  = 0x10335000  (offset = 0x10335000)
  RING_CTRL   = 0x0000f001  (enabled, auto report disabled (0x00), not waiting, semaphore not waiting, length = 0x010 4KB pages)

 - One or more rings were waiting on a semaphore 
  RVSYNC  = 0x00000000
  RVESYNC = 0x00000000
  RBSYNC  = 0x0000cdd1
  BRSYNC  = 0x00076fa9
  BVSYNC  = 0x00000004
  BVESYNC = 0x00000000
  VBSYNC  = 0x0000cdd1
  VRSYNC  = 0x00076fa9
  VVESYNC = 0x00000000

 - Other registers associated with guilty ring - 
  BB_ADDR     = 0x15eacff9  (offset = 0x15eacff8, valid = 0x1)
  BB_STATE    = 0x00000000
  IPEIR       = 0x00000000
  ACTHD       = 0x15eacff8
  EIR         = 0x00000000
  EMR         = 0xffffffff
  ESR         = 0x00000000
  INSTPM      = 0x00006080
  INSTPS      = 0x8000020b
  UHPTR       = 0x00000000

 - Common Slice INST_DONE register -
  Some Common Slice engines are NOT DONE.
   - INSTDONE_CS (=0xffffd390): Not done = SVL, WMFE, WMBE, HiZ, IZ, SBE, RCPBE, DAPB, 

===========================================================================
                 Details for Hung/Stuck ring: MAIN 
===========================================================================
  Last instruction header executed: 0x7a000003  (IPEHR=0x7a000003)
   -> Use AubList to decode:  Aublist -d 0x7a000003
  Some Main engines are NOT DONE.
  INSTDONE  = 0xffdfffff: Not done = 
       - PAGETABLE_RCS_PP_DIR_BASE  = 0x200c0000
       - PAGETABLE_PP_PFIR          = 0x00000000
       - PAGETABLE_PP_PFD           = 0x00000000
       - PAGETABLE_BCS_PP_DIR_BASE  = 0x200c0000
       - PAGETABLE_BCS_PP_DCLV      = 0xffffffff
       - PAGETABLE_BCS_SWCTRL       = 0x00000000
       - PAGETABLE_VCS_PP_DIR_BASE  = 0x10000000
       - PAGETABLE_VCS_PP_DCLV      = 0xffffffff
       - PAGETABLE_VECS_PP_DIR_BASE = 0x10000000
       - PAGETABLE_VECS_PP_DCLV     = 0xffffffff
-----------------------------------
Dumping MAIN ring buffer contents:
  Start Offset:  0x00068dbc
  Head Offset :  0x00068f48   (Adjusted to previous instruction)
  End Offset  :  0x00068f74
-----------------------------------

Offset       Value
------------------------
0x00068dbc : 0x11000001
0x00068dc0 : 0x00002220
0x00068dc4 : 0xffffffff
0x00068dc8 : 0x11000001
0x00068dcc : 0x00002228
0x00068dd0 : 0x200c0000
0x00068dd4 : 0x04000000
0x00068dd8 : 0x0c000000
0x00068ddc : 0x1312d10c
0x00068de0 : 0x00000000
0x00068de4 : 0x00000000
0x00068de8 : 0x00000000
0x00068dec : 0x00000000
0x00068df0 : 0x00000000
0x00068df4 : 0x00000000
0x00068df8 : 0x00000000
0x00068dfc : 0x00000000
0x00068e00 : 0x00000000
0x00068e04 : 0x00000000
0x00068e08 : 0x00000000
0x00068e0c : 0x00000000
0x00068e10 : 0x00000000
0x00068e14 : 0x00000000
0x00068e18 : 0x04000001
0x00068e1c : 0x7a000003
0x00068e20 : 0x01305021
0x00068e24 : 0x00000100
0x00068e28 : 0x00000000
0x00068e2c : 0x00000000
0x00068e30 : 0x7a000003
0x00068e34 : 0x01304498
0x00068e38 : 0x00000080
0x00068e3c : 0x00076fa9
0x00068e40 : 0x00000000
0x00068e44 : 0x01000000
0x00068e48 : 0x11000001
0x00068e4c : 0x00012044
0x00068e50 : 0x00076fa9
0x00068e54 : 0x11000001
0x00068e58 : 0x00022040
0x00068e5c : 0x00076fa9
0x00068e60 : 0x11000001
0x00068e64 : 0x0001a044
0x00068e68 : 0x00076fa9
0x00068e6c : 0x00000000
0x00068e70 : 0x12400001
0x00068e74 : 0x000008d4
0x00068e78 : 0x10345738
0x00068e7c : 0x7a000003
0x00068e80 : 0x01305021
0x00068e84 : 0x00000100
0x00068e88 : 0x00000000
0x00068e8c : 0x00000000
0x00068e90 : 0x7a000003
0x00068e94 : 0x01000c1c
0x00068e98 : 0x00000000
0x00068e9c : 0x00000000
0x00068ea0 : 0x00000000
0x00068ea4 : 0x10400003
0x00068ea8 : 0x00000000
0x00068eac : 0x10201100
0x00068eb0 : 0x00000000
0x00068eb4 : 0x00000000
0x00068eb8 : 0x10400003
0x00068ebc : 0x00000000
0x00068ec0 : 0x10201100
0x00068ec4 : 0x00000000
0x00068ec8 : 0x00000000
0x00068ecc : 0x10400003
0x00068ed0 : 0x00000000
0x00068ed4 : 0x10201100
0x00068ed8 : 0x00000000
0x00068edc : 0x00000000
0x00068ee0 : 0x10400003
0x00068ee4 : 0x00000000
0x00068ee8 : 0x10201100
0x00068eec : 0x00000000
0x00068ef0 : 0x00000000
0x00068ef4 : 0x10400003
0x00068ef8 : 0x00000000
0x00068efc : 0x10201100
0x00068f00 : 0x00000000
0x00068f04 : 0x00000000
0x00068f08 : 0x10400003
0x00068f0c : 0x00000000
0x00068f10 : 0x10201100
0x00068f14 : 0x00000000
0x00068f18 : 0x00000000
0x00068f1c : 0x10400003
0x00068f20 : 0x00000000
0x00068f24 : 0x10201100
0x00068f28 : 0x00000000
0x00068f2c : 0x00000000
0x00068f30 : 0x10400003
0x00068f34 : 0x00000000
0x00068f38 : 0x10201100
0x00068f3c : 0x00000000
0x00068f40 : 0x00000000
0x00068f44 : 0x18800000
0x00068f48 : 0x15ea1e8c  <---- RING HEAD POINTER (PRB0_HEAD)
0x00068f4c : 0x00000000
0x00068f50 : 0x12400001
0x00068f54 : 0x000008d4
0x00068f58 : 0x10345740
0x00068f5c : 0x7a000003
0x00068f60 : 0x01305021
0x00068f64 : 0x00000100
0x00068f68 : 0x00000000
0x00068f6c : 0x00000000
0x00068f70 : 0x7a000003

Aublist command line:
-----------------------
AubList -d 0x11000001 0x00002220 0xffffffff 0x11000001 0x00002228 0x200c0000 0x04000000 0x0c000000 0x1312d10c 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x04000001 0x7a000003 0x01305021 0x00000100 0x00000000 0x00000000 0x7a000003 0x01304498 0x00000080 0x00076fa9 0x00000000 0x01000000 0x11000001 0x00012044 0x00076fa9 0x11000001 0x00022040 0x00076fa9 0x11000001 0x0001a044 0x00076fa9 0x00000000 0x12400001 0x000008d4 0x10345738 0x7a000003 0x01305021 0x00000100 0x00000000 0x00000000 0x7a000003 0x01000c1c 0x00000000 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x10400003 0x00000000 0x10201100 0x00000000 0x00000000 0x18800000 0x15ea1e8c 0x00000000 0x12400001 0x000008d4 0x10345740 0x7a000003 0x01305021 0x00000100 0x00000000 0x00000000 0x7a000003 

* Batch buffer dumps disabled.  Use DEBUG_DUMP_BATCH to enable.


===========================================================================
                          MMIO Register Dump
===========================================================================

MAIN MMIO Registers
-------------------------------------------------
 PAGETABLE_PGTBL_ER                  : 0x00000000
 EXECCONDITION_EXCC                  : 0x00000880
 RING_BUFFER_PRB0_TAIL               : 0x0006e020
 RING_BUFFER_PRB0_HEAD               : 0x01068f4c
 RING_BUFFER_PRB0_START              : 0x10205000
 RING_BUFFER_PRB0_CTL                : 0x000ff001
 CONTROL_RVSYNC                      : 0x00000000
 CONTROL_RVESYNC                     : 0x00000000
 CONTROL_RBSYNC                      : 0x0000cdd1
 RENDERDEBUG_RC_PSMI_CTRL            : 0x00000010
 RENDERDEBUG_RC_PWRCTX_MAXCNT        : 0x00000040
 RENDERDEBUG_IPEHR                   : 0x7a000003
 RENDERDEBUG_INSTDONE_1              : 0xffdfffff
 RENDERDEBUG_INSTPS                  : 0x8000020b
 RENDERDEBUG_ACTHD                   : 0x15eacff8
 RENDERDEBUG_DMA_FADD_P              : 0x15ead180
 CONTROL_INSTDONE_2                  : 0x00000000
 CONTROL_3D_CHICKEN                  : 0x00000000
 CONTROL_3D_CHICKEN2                 : 0x00000000
 CONTROL_3D_CHICKEN3                 : 0x00000000
 CONTROL_3D_CHICKEN4                 : 0x00000000
 IDENTIFICATION_NOPID                : 0x00000000
 CONTROL_HWSTAM                      : 0xffffffff
 CONTROL_MI_MODE                     : 0x00004000
 CONTROL_IMR                         : 0xfffffffe
 CONDITION_EIR                       : 0x00000000
 CONDITION_EMR                       : 0xffffffff
 CONDITION_ESR                       : 0x00000000
 INSTPM                              : 0x00006080
 CONTROL_CMD_CCTL_0                  : 0x00000000
 BATCHBUFF_BB_STATE                  : 0x00000000
 CACHE_MODE_0                        : 0x00006000
 CACHE_MODE_1                        : 0x000001c0
 COMMON_SLICE_INSTDONE               : 0xffffd390
 COMMON_SLICE_CHICKEN1               : 0x00000000
 COMMON_SLICE_CHICKEN2               : 0x00000000
 FF_SLICE_CS_CHICKEN1                : 0x00000000
 RING_BUFFER_UHPTR                   : 0x00000000
 BATCHBUFF_BB_ADDR                   : 0x15eacff9
 BATCHBUFF_BB_PREEMPT_ADDR           : 0x00000000
 RENDER_PR_CTR_CTL                   : 0x00000001
 RENDER_PR_CTR_THRSH                 : 0x00145855
 RING_BUFFER_CCID                    : 0x1312d10d
 RENDER_PR_CTR                       : 0x00000000
 RENDER_CTX_SIZE                     : 0x00000000
 RENDER_CTX_AS_OFFSET                : 0x00000000
 RENDER_CTX_PIPESTATEBASE            : 0x00000000
 SCRATCHPAD_ECOSKPD                  : 0x00000000
 SCRATCHPAD_VFSKPD                   : 0x00000000
 SCRATCHPAD_VFDC                     : 0x00000000
 INTERNAL_CSFLFSM                    : 0x00000500
 INTERNAL_CSFLFLAG                   : 0x50002000
 INTERNAL_CSFLTRK                    : 0x16100000
 INTERNAL_CSCMDOP                    : 0x7a000003
 INTERNAL_CSCMDVLD                   : 0x00000001
 MI_PREDICATE_RESULT_2               : 0x00000000
 OAUNIT_OACONTROL                    : 0x00000000
 OAUNIT_OASTATUS1                    : 0x00000000
 OAUNIT_OASTATUS2                    : 0x0000000b
 OAUNIT_OANOASELECT                  : 0x00000000
 OAUNIT_OAREPORTTRIG4                : 0x00000000
 OAUNIT_OAREPORTTRIG3                : 0x00000000
 OAUNIT_OAREPORTTRIG2                : 0x00000000
 OAUNIT_OAREPORTTRIG1                : 0x00000000
 OAUNIT_OASTARTTRIG4                 : 0x00000000
 OAUNIT_OASTARTTRIG3                 : 0x00000000
 OAUNIT_OASTARTTRIG2                 : 0x00000000
 OAUNIT_OASTARTTRIG1                 : 0x00000000
 OAUNIT_CEC0_0                       : 0x00000000
 OAUNIT_CEC0_1                       : 0x00000000
 OAUNIT_CEC1_0                       : 0x00000000
 OAUNIT_CEC1_1                       : 0x00000000
 OAUNIT_CEC2_0                       : 0x00000000
 OAUNIT_CEC2_1                       : 0x00000000
 OAUNIT_CEC3_0                       : 0x00000000
 OAUNIT_CEC3_1                       : 0x00000000
 OAUNIT_OABUFFER                     : 0x00000000
 RENDER_MI_PREDICATE_SRC0            : 0x00000000
 RENDER_MI_PREDICATE_SRC1            : 0x00000000
 RENDER_MI_PREDICATE_DATA            : 0x00000000
 RENDER_MI_PREDICATE_RESULT          : 0x00000000
 RENDER_3DPRIM_END_OFFSET            : 0x00000000
 RENDER_3DPRIM_START_VERTEX          : 0x00000000
 RENDER_3DPRIM_VERTEX_COUNT          : 0x00000000
 RENDER_3DPRIM_INSTANCE_COUNT        : 0x00000000
 RENDER_3DPRIM_START_INSTANCE        : 0x00000000
 RENDER_3DPRIM_BASE_VERTEX           : 0x00000000
 PAGETABLE_PP_DIR_BASE               : 0x200c0000
 CONTROL_GFX_MODE                    : 0x00002a00
 RENDER_MTCH_CID_RST                 : 0x00000002
 CONTROL_PRBL_SF                     : 0x00000000
 STATUS_RUN_LIST                     : 0x11194001
 STATUS_CTXT_ST_PTR                  : 0x00000000
 STATUS_CTXT_SR_CTL                  : 0x00000000
 STATUS_CTXT_PREMP_DBG               : 0x00000000
 COMMAND_ARB_MODE                    : 0x00000000
 COMMAND_GFX_PEND_TLB_0              : 0x87fd007f
 COMMAND_GFX_PEND_TLB_1              : 0x3f100f00
 COMMAND_ZSHR                        : 0xff00ff00
 COMMAND_CZWMRK                      : 0x00000058
 COMMANDDBG_ECOCHK                   : 0x7a000003
 CONTROL_MIDARB_PRIO                 : 0x00000000
 PAGETABLE_PP_PFIR                   : 0x00000000
 PAGETABLE_PP_PFD                    : 0x00000000
 VECTOR_TLBPEND_VLD0                 : 0x00000000
 VECTOR_TLBPEND_VLD1                 : 0x20408000
 VECTOR_TLBPEND_RDY0                 : 0x00000000
 VECTOR_TLBPEND_RDY1                 : 0x00000000
 VECTOR_MTTLB_VLD0                   : 0x007fffff
 VECTOR_MTTLB_VLD1                   : 0xfffe0000
 VECTOR_VICTLB_VLD0                  : 0x00000000
 VECTOR_VICTLB_VLD1                  : 0x00000000
 VECTOR_RCCTLB_VLD0                  : 0x00000000
 VECTOR_RCCTLB_VLD1                  : 0x00000000
 VECTOR_RCZTLB_VLD0                  : 0x00000000
 VECTOR_RCZTLB_VLD1                  : 0x00000000
 CONTROL_GT_MODE                     : 0x00000080
 PAGETABLE_FBC_RT_BASE_ADDR          : 0x00000000
 GPMUNIT_MESSAGE_BUSRST              : 0x00001f26
 SYSCONFIG_SQCM                      : 0x00000c00
 SYSCONFIG_SQERR                     : 0x00000000
 SYSCONFIG_MFCR                      : 0x00000000
 SYSCONFIG_PERFCNT1L                 : 0x00000000
 SYSCONFIG_PERFCNT1H                 : 0x00000000
 SYSCONFIG_PERFCNT2L                 : 0x00000000
 SYSCONFIG_PERFCNT2H                 : 0x00000000
 SYSCONFIG_PERFMATARIXL              : 0x00000000
 SYSCONFIG_PERFMATARIXH              : 0x00000000
 SYSCONFIG_UCGCTL1                   : 0x00000080
 SYSCONFIG_UCGCTL2                   : 0x00002000
 SYSCONFIG_UCGCTL3                   : 0x00000000
 SYSCONFIG_UCGCTL4                   : 0x02000001
 SYSCONFIG_RCGCTL1                   : 0x00000100
 SYSCONFIG_RCGCTL2                   : 0x00000000
 SYSCONFIG_IFDIMCTL                  : 0x00000000
 SYSCONFIG_GDRST                     : 0x00000000
 SYSCONFIG_RSTCTL                    : 0x00000000
 SYSCONFIG_MISCCPCTL                 : 0x000003fd
 SYSCONFIG_PSMIBASE                  : 0xcba5f0c1
 SYSCONFIG_PSMICTRL                  : 0x00000000
 SYSCONFIG_PSMISTATUS                : 0x00000000
 CONTROL_TD_CTL2                     : 0x00000000
 CONTROL_TD_VF_VS_EMSK               : 0x00000000
 CONTROL_TD_GS_EMSK                  : 0x00000000
 CONTROL_TD_HS_EMSK                  : 0x00000000
 CONTROL_TD_DS_EMSK                  : 0x00000000
 CONTROL_TD_WIZ_EMSK                 : 0x00000000
 CONTROL_TD_TS_EMSK                  : 0x00000000

MMIO Offset:
------------

0x00002000  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002020  :  0x00000000 0x00000000 0x00000880 0x00000000 0x0006e020 0x01068f4c 0x10205000 0x000ff001 
0x00002040  :  0x00000000 0x0000cdd1 0x00000000 0x00000000 0x00000010 0x00000040 0x00000000 0x00000000 
0x00002060  :  0x00000000 0x00000000 0x7a000003 0xffdfffff 0x8000020b 0x15eacff8 0x15ead180 0x00000000 
0x00002080  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00004000 
0x000020a0  :  0x28a01010 0x00000000 0xfffffffe 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 
0x000020c0  :  0x00006080 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000020e0  :  0x00000000 0x00000000 0x00000001 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002100  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002120  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002140  :  0x15eacff9 0x130c3308 0x00000000 0x00000000 0x15ea1e8c 0x0000b158 0x00000001 0x00000000 
0x00002160  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000001 0x00145855 
0x00002180  :  0x1312d10d 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000021a0  :  0x00000000 0x00000000 0xe54d430d 0x0000126a 0x00000000 0x000029c0 0x00008580 0x00000000 
0x000021c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000021e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002200  :  0x00000500 0x50002000 0x16100000 0x7a000003 0x00000001 0x00000000 0x15eacff8 0x00000000 
0x00002220  :  0xffffffff 0x00000000 0x200c0000 0x00000002 0x00000000 0x11194001 0x00000000 0x00000000 
0x00002240  :  0x0001e000 0x00000000 0x00000000 0xa1000000 0x47a16103 0xd52c3080 0x33a3282b 0x1621e502 
0x00002260  :  0x14a10141 0x73425294 0x0001c224 0x96115ee3 0x0001e4a7 0x111948ac 0x1501e097 0xc02038e4 
0x00002280  :  0x1101d267 0x081c6012 0xf461d403 0x19000100 0x00000000 0x00000000 0x00000000 0x00002a00 
0x000022a0  :  0x00000000 0x10000000 0x0800000b 0x00400000 0x00a00000 0x00000000 0x00000600 0x00000000 
0x000022c0  :  0x10000003 0x00000000 0x4c754a70 0x00000000 0x00000000 0x00000000 0x130f1fd9 0x00000000 
0x000022e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x4c67beb0 0x00000000 0x130bb63e 0x00000000 
0x00002300  :  0x00000000 0x00000000 0x00000000 0x00000000 0x001d23df 0x00000000 0x0009b6a7 0x00000000 
0x00002320  :  0x000f43dd 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x0009b6a5 0x00000000 
0x00002340  :  0x00099503 0x00000000 0x98dd0920 0x00000000 0x261ad617 0x00000000 0x268226a0 0x00000005 
0x00002360  :  0x00000000 0x00000000 0x0000000b 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002380  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000023a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000023c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000023e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002400  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002420  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002440  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002460  :  0x0022c000 0x2000000b 0x2a000011 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002480  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000024a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000024c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000024e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002500  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002520  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002540  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002560  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002580  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000025a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000025c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000025e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00000000 
0x00002600  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002620  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002640  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002660  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002680  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000026a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000026c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000026e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002700  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002720  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002740  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002760  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00002780  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000027a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000027c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000027e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 



BLIT MMIO Registers
-------------------------------------------------
 EXECCONDITION_BCS_EXCC              : 0x00000000
 RING_BUFFER_BCS_RING_TAIL           : 0x00012610
 RING_BUFFER_BCS_RING_HEAD           : 0x0e6118d8
 RING_BUFFER_BCS_RING_START          : 0x10305000
 RING_BUFFER_BCS_RING_CTL            : 0x0001f401
 CONTROL_BRSYNC                      : 0x00076fa9
 CONTROL_BVSYNC                      : 0x00000004
 CONTROL_BVESYNC                     : 0x00000000
 BLTDEBUG_RC_PSMI_CTRL               : 0x00000010
 BLTDEBUG_RC_PWRCTX_MAXCNT           : 0x0000000a
 BLTDEBUG_BCS_IPEIR                  : 0x00000000
 BLTDEBUG_BCS_IPEHR                  : 0x0b140001
 BLTDEBUG_BCS_INSTDONE               : 0xfffffffe
 BLTDEBUG_BCS_ACTHD                  : 0x0e6118e4
 BLTRDEBUG_BCS_DMA_FADD              : 0x103168d8
 IDENTIFICATION_BCS_NOPID            : 0x00000000
 CONTROL_BCS_HWSTAM                  : 0xffffffff
 CONTROL_BCS_MI_MODE                 : 0x00000200
 CONTROL_BCS_GAC_MODE                : 0x00000000
 CONTROL_BCS_IMR                     : 0xffbfffff
 CONTROL_BCS_EIR                     : 0x00000000
 CONTROL_BCS_EMR                     : 0xffffffff
 CONTROL_BCS_ESR                     : 0x00000000
 BLT_BCS_INSTPM                      : 0x00000000
 RING_BUFFER_BCS_UHPTR               : 0x00000000
 BATCHBUFF_BCS_BB_ADDR               : 0x00000000
 BLT_RING_BCS_RCCID                  : 0x00000000
 BLT_RING_BCS_RNCID                  : 0x00000000
 SCRATCHPAD_BCS_ECOSKPD              : 0x00000000
 PAGETABLE_BCS_SWCTRL                : 0x00000000
 PAGETABLE_BCS_PP_DCLV               : 0xffffffff
 PAGETABLE_BCS_PP_DIR_BASE           : 0x200c0000
 CONTROL_BLT_MODE                    : 0x00000200
 BLT_TIMESTAMP                       : 0x26822ecb
 BLT_GAC_GAB                         : 0x00000000
 COMMAND_BCS_HWS_PGA                 : 0x10203000
 BLT_GAB_REPORT                      : 0x00000000
 BLTVECTOR_TLBPEND_VLD0              : 0x00000000
 BLTVECTOR_TLBPEND_RDY0              : 0x00000000
 BLTVECTOR_BCS_TLB_VLD               : 0x00000000
 BLTVECTOR_BLB_TLB_VLD               : 0x00000000
 BLTVECTOR_CTX_TLB_VLD               : 0x00000000
 BLTVECTOR_PD_TLB_VLD                : 0x00000000
 SPRITE_DVSASCALE                    : 0x00000000
 SPRITE_DVSBSURFLIVE                 : 0x00000000
 SPRITE_DVSBSCALE                    : 0x00000000

MMIO Offset:
------------

0x00022000  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022020  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00012610 0x0e6118d8 0x10305000 0x0001f401 
0x00022040  :  0x00076fa9 0x00000004 0x00000000 0x00000000 0x00000010 0x0000000a 0x00000000 0x00000000 
0x00022060  :  0x00000000 0x00000000 0x0b140001 0xfffffffe 0x00000000 0x0e6118e4 0x103168d8 0x00000000 
0x00022080  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00000200 
0x000220a0  :  0x00000000 0x00000000 0xffbfffff 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 
0x000220c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000220e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022100  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022120  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022140  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022160  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00150000 
0x00022180  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000221a0  :  0x00005400 0x00005500 0x00000402 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000221c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000221e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022200  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022220  :  0xffffffff 0x00000000 0x200c0000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022240  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022260  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022280  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000200 
0x000222a0  :  0x00080000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000222c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000222e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022300  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022320  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022340  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x2682345e 0x00000005 
0x00022360  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022380  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000223a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000223c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000223e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022400  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022420  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00012610 0x0e6118d8 0x10305000 0x0001f401 
0x00022440  :  0x00076fa9 0x00000004 0x00000000 0x00000000 0x00000010 0x0000000a 0x00000000 0x00000000 
0x00022460  :  0x00000000 0x00000000 0x0b140001 0xfffffffe 0x00000000 0x0e6118e4 0x103168d8 0x00000000 
0x00022480  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00000200 
0x000224a0  :  0x00000000 0x00000000 0xffbfffff 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 
0x000224c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000224e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022500  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022520  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022540  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022560  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00150000 
0x00022580  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000225a0  :  0x00005400 0x00005500 0x00000402 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000225c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000225e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022600  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022620  :  0xffffffff 0x00000000 0x200c0000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022640  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022660  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022680  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000200 
0x000226a0  :  0x00080000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000226c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000226e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022700  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022720  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022740  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022760  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00022780  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000227a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000227c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000227e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 



MEDIA MMIO Registers
-------------------------------------------------
 COMMAND_VCS_HWS_PGA                 : 0x10202000
 RING_BUFFER_VCS_RING_TAIL           : 0x00000168
 RING_BUFFER_VCS_RING_HEAD           : 0x00000168
 RING_BUFFER_VCS_RING_START          : 0x10325000
 RING_BUFFER_VCS_RING_CTL            : 0x0000f001
 CONTROL_VBSYNC                      : 0x0000cdd1
 CONTROL_VRSYNC                      : 0x00076fa9
 CONTROL_VVESYNC                     : 0x00000000
 MEDIADEBUG_RC_PSMI_CTRL             : 0x00000010
 MEDIADEBUG_RC_PWRCTX_MAXCNT         : 0x0000000a
 MEDIADEBUG_VCS_IPEIR                : 0x00000000
 MEDIADEBUG_VCS_IPEHR                : 0x00000000
 MEDIADEBUG_VCS_INSTDONE             : 0xfffffffe
 MEDIADEBUG_VCS_ACTHD                : 0x00000168
 MEDIADEBUG_VCS_DMA_FADD             : 0x10325168
 IDENTIFICATION_VCS_NOPID            : 0x00000000
 CONTROL_VCS_HWSTAM                  : 0xffffffff
 CONTROL_VCS_MI_MODE                 : 0x00000200
 CONTROL_VCS_GAC_MODE                : 0x00000000
 CONTROL_VCS_IMR                     : 0xffffefff
 CONTROL_VCS_EIR                     : 0x00000000
 CONTROL_VCS_EMR                     : 0xffffffff
 CONTROL_VCS_ESR                     : 0x00000000
 MEDIA_VCS_INSTPM                    : 0x00000000
 BATCHBUFF_VCS_BB_STATE              : 0x00000000
 STATUS_VCS_CTXT_SR_CTL              : 0x00000000
 BATCHBUFF_VCS_BB_ADDR               : 0x00000000
 MEDIA_VCS_CNTR                      : 0xffffffff
 MEDIA_VCS_THRSH                     : 0x00150000
 MEDIA_RING_VCS_RCCID                : 0x00000000
 MEDIA_RING_VCS_RNCID                : 0x00000000
 SCRATCHPAD_VCS_ECOSKPD              : 0x00000000
 PAGETABLE_VCS_PP_DCLV               : 0xffffffff
 PAGETABLE_VCS_PP_DIR_BASE           : 0x10000000
 MEDIA_TIMESTAMP                     : 0x26823cef
 MFC_VIN_AVD_ERROR_FLAGS             : 0x00000000
 MFC_VIN_AVD_ERROR                   : 0x00000000
 MFC_BITSTREAM_BYTECOUNT_SLICE       : 0x00000000
 MFC_BITSTREAM_SE_BITCOUNT_SLICE     : 0x00000000
 MFC_AVC_CABAC_INSERTION_COUNT       : 0x00000000
 MFC_AVC_MINSIZE_PADDING_COUNT       : 0x00000000
 MFC_IMAGE_STATUS_MASK               : 0x00000000
 MFC_IMAGE_STATUS_CONTROL            : 0x00000000
 MFC_BITSTREAM_BYTECOUNT_FRAME       : 0x00000000
 MFC_BITSTREAM_SE_BITCOUNT_FRAME     : 0x00000000
 MFC_AVC_CABAC_BIN_COUNT_FRAME       : 0x00000000
 DEBUG_MFC_BSP_DEBUG0                : 0x00000000
 DEBUG_MFC_BSP_DEBUG1                : 0x00000000
 DEBUG_MFC_BSP_DEBUG2                : 0x00000000
 DEBUG_MFC_VIN_DEBUG0                : 0x00000000
 DEBUG_MFC_VIN_DEBUG1                : 0x00000000
 DEBUG_MFC_VIN_DEBUG2                : 0x00000000
 DEBUG_MFC_VIN_DEBUG3                : 0x00000000
 MEDIACONTROL_TLBPEND                : 0x00000000
 MEDIA_GAC_GAB                       : 0x00000000
 MEDIACOMMANDDBG_ECOCHK              : 0x00000000
 MEDIACOMMANDDBG_ERROR               : 0x00000000
 MEDIAVECTOR_TLBPEND_VLD0            : 0x00000000
 MEDIAVECTOR_TLBPEND_VLD1            : 0x00000000
 MEDIAVECTOR_TLBPEND_RDY0            : 0x00000000
 MEDIAVECTOR_TLBPEND_RDY1            : 0x00000000
 MEDIAVECTOR_TLB064_VLD0             : 0x00000000
 MEDIAVECTOR_TLB064_VLD1             : 0x00000000
 MEDIAVECTOR_TLB132_VLD0             : 0x00000000
 MEDIAVECTOR_TLB132_VLD1             : 0x00000000
 MEDIAVECTOR_TLB232_VLD0             : 0x00000000
 MEDIAVECTOR_TLB232_VLD1             : 0x00000000
 MEDIAVECTOR_TLB304_VLD0             : 0x00000000
 MEDIAVECTOR_TLB304_VLD1             : 0x00000000

MMIO Offset:
------------

0x00012000  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012020  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000168 0x00000168 0x10325000 0x0000f001 
0x00012040  :  0x0000cdd1 0x00076fa9 0x00000000 0x00000000 0x00000010 0x0000000a 0x00000000 0x00000000 
0x00012060  :  0x00000000 0x00000000 0x00000000 0xfffffffe 0x00000000 0x00000168 0x10325168 0x00000000 
0x00012080  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00000200 
0x000120a0  :  0x00000000 0x00000000 0xffffefff 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 
0x000120c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000120e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012100  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012120  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012140  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012160  :  0x00000000 0xffffffff 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00150000 
0x00012180  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000121a0  :  0x00004000 0x00004480 0x00040e02 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000121c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000121e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012200  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012220  :  0xffffffff 0x00000000 0x10000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012240  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012260  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012280  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000300 
0x000122a0  :  0x00080000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000122c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000122e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012300  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012320  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012340  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x26824674 0x00000005 
0x00012360  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012380  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000123a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000123c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000123e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012400  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012420  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012440  :  0x00000000 0x00000000 0x00100000 0x00000000 0x00000000 0x00200020 0x00000000 0x00000000 
0x00012460  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012480  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000124a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000124c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000124e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012500  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012520  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012540  :  0x00000000 0x00000000 0x00100000 0x00000000 0x00000000 0x00200020 0x00000000 0x00000000 
0x00012560  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012580  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000125a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000125c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000125e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012600  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012620  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012640  :  0x00000000 0x00000000 0x00100000 0x00000000 0x00000000 0x00200020 0x00000000 0x00000000 
0x00012660  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012680  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000126a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000126c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000126e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012700  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012720  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012740  :  0x00000000 0x00000000 0x00100000 0x00000000 0x00000000 0x00200020 0x00000000 0x00000000 
0x00012760  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x00012780  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000127a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000127c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x000127e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 



VEBOX MMIO Registers
-------------------------------------------------
 COMMAND_VECS_HWS_PGA                : 0x10204000
 RING_BUFFER_VECS_RING_TAIL          : 0x000000f0
 RING_BUFFER_VECS_RING_HEAD          : 0x000000f0
 RING_BUFFER_VECS_RING_START         : 0x10335000
 RING_BUFFER_VECS_RING_CTL           : 0x0000f001
 CONTROL_VEBSYNC                     : 0x0000cdd1
 CONTROL_VERSYNC                     : 0x00076fa9
 CONTROL_VEVSYNC                     : 0x00000004
 VEBOXDEBUG_RC_PSMI_CTRL             : 0x00000010
 VEBOXDEBUG_RC_PWRCTX_MAXCNT         : 0x0000000a
 VEBOXDEBUG_VECS_IPEIR               : 0x00000000
 VEBOXDEBUG_VECS_IPEHR               : 0x00000000
 VEBOXDEBUG_VECS_INSTDONE            : 0xfffffffe
 VEBOXDEBUG_VECS_ACTHD               : 0x000000f0
 VEBOXDEBUG_VECS_DMA_FADD            : 0x103350f0
 IDENTIFICATION_VECS_NOPID           : 0x00000000
 CONTROL_VECS_HWSTAM                 : 0xffffffff
 CONTROL_VECS_MI_MODE                : 0x00000200
 CONTROL_VECS_IMR                    : 0xfffffbff
 CONTROL_VECS_EIR                    : 0x00000000
 CONTROL_VECS_EMR                    : 0xffffffff
 CONTROL_VECS_ESR                    : 0x00000000
 VEBOX_VECS_INSTPM                   : 0x00000000
 BATCHBUFF_VECS_BB_STATE             : 0x00000000
 BATCHBUFF_VECS_BB_ADDR              : 0x00000000
 PAGETABLE_VECS_PP_DCLV              : 0xffffffff
 PAGETABLE_VECS_PP_DIR_BASE          : 0x10000000
 VEBOX_TIMESTAMP                     : 0x26824f72

MMIO Offset:
------------

0x0001a000  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a020  :  0x00000000 0x00000000 0x00000000 0x00000000 0x000000f0 0x000000f0 0x10335000 0x0000f001 
0x0001a040  :  0x0000cdd1 0x00076fa9 0x00000004 0x00000000 0x00000010 0x0000000a 0x00000000 0x00000000 
0x0001a060  :  0x00000000 0x00000000 0x00000000 0xfffffffe 0x00000000 0x000000f0 0x103350f0 0x00000000 
0x0001a080  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00000200 
0x0001a0a0  :  0x00000000 0x00000000 0xfffffbff 0x00000000 0x00000000 0xffffffff 0x00000000 0x00000000 
0x0001a0c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a0e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a100  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a120  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a140  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a160  :  0x00000000 0xffffffff 0x00000000 0x00000000 0x00000000 0x00000000 0xffffffff 0x00150000 
0x0001a180  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a1a0  :  0x00004800 0x00004900 0x00040002 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a1c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a1e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a200  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a220  :  0xffffffff 0x00000000 0x10000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a240  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a260  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a280  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000200 
0x0001a2a0  :  0x00080000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a2c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a2e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a300  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a320  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a340  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x2682547e 0x00000005 
0x0001a360  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a380  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a3a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a3c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a3e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a400  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a420  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a440  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a460  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a480  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a4a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a4c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a4e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a500  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a520  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a540  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a560  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a580  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a5a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a5c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a5e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a600  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a620  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a640  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a660  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a680  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a6a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a6c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a6e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a700  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a720  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a740  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a760  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a780  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a7a0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a7c0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 
0x0001a7e0  :  0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 0x00000000 



OTHER MMIO Registers
-------------------------------------------------
 SYSCONFIG_DFSM                      : 0x20000002
 SYSCONFIG_DSSM                      : 0x00000001
 SYSCONFIG_DPFC_CONTROL_SA           : 0x20000000
 SYSCONFIG_DPFC_CPU_FENCE_OFFSET     : 0x00000000
 SYSCONFIG_TILECTL                   : 0x00000000
 SYSCONFIG_MISC0                     : 0x00000000
 SYSCONFIG_MGGC                      : 0x00000000
 SYSCONFIG_MTOLUD                    : 0x7fa00001
 SYSCONFIG_GFX_FLSH_CNTL             : 0x00000211
 SYSCONFIG_MTOUUD                    : 0x7f600001
 SYSCONFIG_MBDSM                     : 0x7ba00001
 SYSCONFIG_MBGSM                     : 0x7b800001
 SYSCONFIG_GTVTD0                    : 0xffffffff
 SYSCONFIG_GTVTD1                    : 0xffffffff
 SYSCONFIG_GTVTD2                    : 0xffffffff
 SYSCONFIG_GTVTD3                    : 0xffffffff
 SYSCONFIG_GTVTD4                    : 0xffffffff
 SYSCONFIG_GTVTD5                    : 0xffffffff
 SYSCONFIG_GTVTD6                    : 0xffffffff
 SYSCONFIG_GTVTD7                    : 0xffffffff
 SYSCONFIG_GTVTD8                    : 0xffffffff
 SYSCONFIG_GTVTD9                    : 0xffffffff
 SYSCONFIG_GTVTD10                   : 0xffffffff
 SYSCONFIG_GTVTD11                   : 0xffffffff
 SYSCONFIG_GTVTD12                   : 0xffffffff
 SYSCONFIG_GTVTD13                   : 0xffffffff
 SYSCONFIG_GTVTD14                   : 0xffffffff
 SYSCONFIG_GTVTD15                   : 0xffffffff



===========================================================================
                            End Hang Analysis
===========================================================================
```

**Comment by gvr1421** [comment]
+

**Comment by Th3NightHawk** [comment]
It's been 2 days with no Dota... feeling withdrawals... HELP?!?!?

New update today, game still crashes.

**Comment by gdrewb-valve** [comment]
In situations like this where the GPU hangs up it could be bad hardware, bad OpenGL software or Dota.  If we can repro the situation we can try and figure out what's going on but as we aren't seeing widespread reports of this I'm not sure we'll be able to turn up much.

**Comment by amisala** [comment]
Hi everyone, 

I am having the same issue - keeps on crashing since last Saturday, so it's 8 days already. Strangely, a bot match won't crash for me, neither will spectacting a game. Hero demo, however, does crash. Everything else crashes, too. 
I am no expert on this matter, so I only attached a printscreen of my console, don't know what else I could submit. 
It's a shame we didn't even get a chance to play Dark Moon and after purchasing our Battle Passes, we can't even use them.

Have a nice day,
Adrian 

<img width="1280" alt="snimka obrazovky 2017-02-05 o 19 28 00" src="https://cloud.githubusercontent.com/assets/25569359/22628729/6b6531be-ebd9-11e6-9058-7a6322338a4a.png">

**Comment by Th3NightHawk** [comment]
Is there something we can do to try narrow down the cause?

**Comment by sandyre** [comment]
Expecting same issue, just want to mention that OSX crash is possibly connected to OSX 10.12.4 (16E144f) Beta, so it might be a bug in Beta version of OSX.

**Comment by amisala** [comment]
I'll try downgrading from Beta and will get back to you.

**Comment by amisala** [comment]
Downgraded to 10.12.3 (16D32), game runs smoothly! 
At last a solution that works after 9 days.

**Comment by gdrewb-valve** [comment]
Great, we'll wait until @Th3NightHawk can confirm and then we'll close this out.

**Comment by Th3NightHawk** [comment]
Thanks guys, I'll try downgrade and feedback tonight.

**Comment by gvr1421** [comment]
i have 10.12.3, but still dota crashes

**Comment by gdrewb-valve** [comment]
@gvr1421 are you referring to your issue 1162?  If so you have a different problem than this one.

**Comment by gvr1421** [comment]
@gdrewb-valve i think its not different, because initially the title of topic was "mac os dota crashing". Some1 changed it. I ve left thee video^ how is it happening that dota crashes

**Comment by gdrewb-valve** [comment]
@gvr1421 in the bottom of the comments on 1162 you say "Hello again) as i said dota works", is that accurate?

**Comment by gvr1421** [comment]
@gdrewb-valve it works after i downloaded nvidia web driver. It works only if i use this driver. And i have 15 fps in game6 so its impossible to play normal=( When i use default driver, dota doesn't work.

**Comment by gdrewb-valve** [comment]
Your driver problem would then be a separate issue from the one here that was fixed by not using 10.12.4, wouldn't it?  I'm trying to understand why you think you have the same problem.  When you use the default driver are you getting GPU restart files in the DiagnosticReports folder?

**Comment by gvr1421** [comment]
@gdrewb-valve "My Dota 2 client keeps crashing without any errors once I launch into a game. It's fine in the menu but even going into the Demo Hero mode crashes the game. " its this issue. so i have the same. https://youtu.be/HrWiYFwB7qs

**Comment by gdrewb-valve** [comment]
Not all crashes are the same.  This one in particular seems to be related to 10.12.4 and will be closed when Th3NightHawk confirms.  Your issue at 1162, which appears to be related to the OpenGL driver you're using, will stay open.

**Comment by gvr1421** [comment]
ok

**Comment by sandyre** [comment]
macOS 10.12.4 Public Beta 2 (16E154a), problem stays.

**Comment by gdrewb-valve** [comment]
Closing as this appears to be a macOS issue and not a Dota issue.

**Comment by Th3NightHawk** [comment]
Hi guys

Sorry for the delayed feedback, was in a car accident earlier in the week so I had other crap to take care of.

Just wanted to confirm that reverting to macOS 10.12.3 resolves the issue.
